### PR TITLE
Update Meshery Filters

### PR DESCRIPTION
**Pull request by pottekkat**
**Description**

Added GitHub walker for patterns and updated the endpoints.

This implementation is similar to the current implementation of patterns.

Endpoints added-

`/api/experimental/filter` `POST` `GET`
`/api/experimental/filter/{id}` `GET`
`/api/experimental/filter/{id}` `DELETE`

Not added yet-

`/api/experimental/filter/deploy`

- [x] Yes, I signed my commits.

### CODE REVIEW

File: models/default_local_provider.go [hunk 1]
```diff
@@ -546,46 +546,50 @@ func (l *DefaultLocalProvider) DeleteMesheryFilter(req *http.Request, filterID s
...
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("file not found")
-	}
 
-	respJSON := map[string]interface{}{}
+	// Check if hostname is github
+	if parsedURL.Host == "github.com" {
+		parsedPath := strings.Split(parsedURL.Path, "/")
+		if len(parsedPath) < 3 {
+			return nil, fmt.Errorf("malformed URL: url should be of type github.com/<owner>/<repo>/[branch]")
```
Comment (leecalcote) [inline models/default_local_provider.go:560]: @kumarabd at what point can we start to use meshkit for reporting errors?

**Comment by leecalcote** [comment]
- The API endpoints in this PR match with the endpoints and verbs defined [here](https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399) ✔️ 
- /experimental for now - ok.
- Are "filters" registered as a capability of remote providers? Where is this defined?

**Comment by leecalcote** [comment]
> @leecalcote filters have also been added as a capability to the remote provider in this accompanying PR [layer5io/meshery-cloud#78](https://github.com/layer5io/meshery-cloud/pull/78)

Very good. Can you highlight this extension point (in this PR) for me?

**Review by utkarsh-pro — APPROVED** [review]

**Comment by navendu-pottekkat** [comment]
@leecalcote @utkarsh-pro please also check the PR on meshery-cloud https://github.com/layer5io/meshery-cloud/pull/78

**Pull request merged**